test/framework/clustervalidator.go

```diff
@@ -3,277 +3,104 @@ package framework
-	c.validations = append(c.validations, retriableValidation{validation, backoffPeriod, maxRetries})
-}
-
-// WithValidations registers multiple validations to the ClusterValidator that will be run when Validate is called.
-func (c *ClusterValidator) WithValidations(validation ...ClusterValidation) {
-	for _, v := range validation {
-		c.validations = append(c.validations, retriableValidation{v, 5 * time.Second, 30})
-	}
-}
-
-// WithWorkloadClusterValidations registers a validation set for a workload cluster.
-func (c *ClusterValidator) WithWorkloadClusterValidations() {}
-
-// WithExpectedObjectsExist registers a set of validations for the existence of various cluster objects.
-func (c *ClusterValidator) WithExpectedObjectsExist() {
-	c.WithValidation(validateClusterReady, 5*time.Second, 60)
-	c.WithValidation(validateEKSAObjects, 5*time.Second, 60)
-	c.WithValidation(validateControlPlaneNodes, 5*time.Second, 120)
-	c.WithValidation(validateWorkerNodes, 5*time.Second, 120)
-	c.WithValidation(validateCilium, 5*time.Second, 60)
-}
-
-// WithClusterDoesNotExist registers a validation to check that a cluster does not exist or has been deleted.
-func (c *ClusterValidator) WithClusterDoesNotExist() {
-	c.WithValidation(validateClusterDoesNotExist, 5*time.Second, 60)
-}
-
-// Validate runs through the set registered validations and returns an error if any of them fail after a number of retries.
-func (c *ClusterValidator) Validate(ctx context.Context) error {
-	for _, v := range c.validations {
-		err := retrier.Retry(v.maxRetries, v.backoffPeriod, func() error {
-			return v.validation(ctx, c.Config)
-		})
-		if err != nil {
-			return fmt.Errorf("validation failed %v", err)
-		}
+// WithValidationsForExpectedObjects registers a set validations for the existence of various cluster objects.
+func WithValidationsForExpectedObjects(cv *clusterf.StateValidator) *clusterf.StateValidator {
+	mediumRetier := retrier.NewWithMaxRetries(60, 5*time.Second)
+	longRetier := retrier.NewWithMaxRetries(120, 5*time.Second)
+	cv.WithValidations(
+		clusterf.RetriableStateValidation(mediumRetier, csvalidations.ValidateClusterReady),
+		clusterf.RetriableStateValidation(mediumRetier, csvalidations.ValidateEKSAObjects),
+		clusterf.RetriableStateValidation(longRetier, csvalidations.ValidateControlPlaneNodes),
+		clusterf.RetriableStateValidation(longRetier, csvalidations.ValidateWorkerNodes),
+		clusterf.RetriableStateValidation(mediumRetier, csvalidations.ValidateCilium),
+	)
+	return cv
+}
+
+// WithValidationsForClusterDoesNotExist registers validations to check that a cluster does not exist or has been deleted.
+func WithValidationsForClusterDoesNotExist(cv *clusterf.StateValidator) *clusterf.StateValidator {
+	cv.WithValidations(
+		clusterf.RetriableStateValidation(retrier.NewWithMaxRetries(60, 5*time.Second), csvalidations.ValidateClusterDoesNotExist),
+	)
+	return cv
+}
+
+func (e *ClusterE2ETest) initClusterStateConfig(ctx context.Context) {
```

Comment:
I would advise against this `init` method
I would just have a `clusterStateConfig(ctx)` that checks if cluster state config is memoized and build it if it's not

Reply:
Curious, what is your reasoning behind being strongly against this `init` method? I could simply build when `ValidateClusterState` is called, but I was going for keeping the spec in the `clusterValidatorConfig` up to date with the in the running cluster when the API is used.